### PR TITLE
chore(dependency-mgmt): Migrate dependency management jobs to Github

### PR DESCRIPTION
**Pull request by venkkatesh-sekar**

### CODE REVIEW

File: .github/workflows/ci-pr-only.yml [hunk 1]
```diff
@@ -101,3 +101,45 @@ jobs:
         id: pre-commit
         shell: bash
         run: ./gitlab-ci/src/ci-scripts/pre-commit.sh
+  dependencies-check:
```
Comment (marko-k0) [inline .github/workflows/ci-pr-only.yml:104]: @venkkatesh-sekar, you need to define these under `.github/workflows-source`. We have that due to DRY reasons.

File: gitlab-ci/src/dependencies/integration/github/github_api.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        # Check if dependency management comment already exists
+        update_comment = False
+        for comment in comments:
+            if comment.body.startswith(DELTA_HEADER):
+                update_comment = True
+                break
+
+        if update_comment:
+            comment.edit(comment_body)
```
[the remark this answers is not in the document]
Comment (venkkatesh-sekar) [reply gitlab-ci/src/dependencies/integration/github/github_api.py:54]: Python doesn't have block scoping afaik, so `comment` should still be in scope.

File: .github/workflows/ci-pr-only.yml (hunk 1 above)
Comment (venkkatesh-sekar) [reply .github/workflows/ci-pr-only.yml:104]: Thanks @marko-k0, I moved them

File: .github/workflows-source/ci-pr-only.yml [hunk 3]
```diff
@@ -107,3 +107,35 @@ jobs:
...
+      JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+      SLACK_PSEC_BOT_OAUTH_TOKEN: ${{ secrets.SLACK_PSEC_BOT_OAUTH_TOKEN }}
+    steps:
+      - <<: *checkout
+      - <<: *before-script
+      - name: Setup python deps
+        id: setup-python-deps
+        shell: bash
+        run: |
+          pip3 install --ignore-installed -r requirements.txt
```
Comment (marko-k0) [inline .github/workflows-source/ci-pr-only.yml:133]: Seems a bit off to have pip3 install in every PR. Would it make sense to put this in the `ic-build` image? Or maybe use `PIP_DOWNLOAD_CACHE` in `/cache/pip`?

Comment (venkkatesh-sekar) [reply .github/workflows-source/ci-pr-only.yml:133]: AFAIR, we used to already have it in the `ic-build` image but @cgundy made a PR long back to move this to outside of the image.

Comment (cgundy) [reply .github/workflows-source/ci-pr-only.yml:133]: Yeah we don't have python packages pre-installed in the image and we install it this way for other CI jobs too. We can discuss later if it makes sense to do a refactor but from what I remember we wanted to keep the images as small as possible.

Comment (cgundy) [reply .github/workflows-source/ci-pr-only.yml:133]: See reference here for why we removed them from the docker image: https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/15567

```
Bazel can run into conflicts when the dependencies are pre-installed in the CI image.
```

**Review by marko-k0 — APPROVED** [review]

**Pull request merged**